test/cpp/jit/backend_with_compiler.cpp

```diff
@@ -0,0 +1,159 @@
+// inference.
+
+// Impletation details:
+//
+// Compilation
+// 1. A backend with minimum compilation features, "backend_with_compiler_demo"
+// is added.
+// 2. The compilation happens AOT in the pre_process function registered to this
+// backend.
+// 3. Compiled results are stored in a string blob for each method. They are
+// serialized to the lowered module with __get_state__ function.
+// 4. Error message with model source code is thrown, for features not handled
+// by the backend compiler.
+//
+// Runtime
+// 1. The compiled blob is loaded in __set_state__ method.
+// 2. The compile function of the backend: parse the preprocessed blob to the
+// format (a list of tokens) that the backend can understand.
+// 3. The execute function of the backend executes the specified method
+// (handle).
+
+namespace {
+std::vector<std::string> parseMethodHandle(const std::string& blob) {
+  std::vector<std::string> result;
+  std::stringstream s_stream(blob);
+  while (s_stream.good()) {
+    std::string substr;
+    getline(s_stream, substr, ',');
+    result.push_back(substr);
+  }
+  return result;
+}
+} // namespace
+
+class BackendWithCompiler : public PyTorchBackendInterface {
+ public:
+  // Constructor.
+  explicit BackendWithCompiler() {}
+  virtual ~BackendWithCompiler() = default;
+
+  // Since the actual compilation is done AOT,
+  c10::impl::GenericDict compile(
+      c10::IValue processed,
+      c10::impl::GenericDict method_compile_spec) override {
+    //    return processed.toGenericDict();
+    auto dict = processed.toGenericDict();
+    auto handles = c10::Dict<std::string, std::vector<std::string>>();
+    for (const auto& kv : dict) {
+      auto tokens = parseMethodHandle(kv.value().toStringRef());
+      handles.insert(kv.key().toStringRef(), tokens);
+    }
+    return c10::impl::toGenericDict(handles);
+  }
+
+  c10::impl::GenericList execute(
+      c10::IValue handle,
+      c10::impl::GenericList inputs) override {
+    //    TORCH_INTERNAL_ASSERT(handle.isString());
+    TORCH_INTERNAL_ASSERT(inputs.size() == 2);
+    c10::IValue val0 = inputs[0];
+    at::Tensor x = val0.toTensor();
+    c10::IValue val1 = inputs[1];
+    at::Tensor h = val1.toTensor();
+
+    c10::List<at::Tensor> output_list;
+    double scalar_val = 1.0;
+    for (const auto& token : handle.toList()) {
+      IValue val = token;
+      auto instruction = std::string(IValue(token).toStringRef());
+      double const_val = 1.0;
+      if (instruction.rfind("prim::Constant", 0) == 0) {
+        TORCH_CHECK(
+            instruction.size() > 15,
+            "Constant value is expected in ",
+            instruction);
+        auto sub = instruction.substr(15);
+        const_val = stod(sub);
+      } else if (token == "aten::add") {
+        output_list.emplace_back(x.add_(h, const_val));
+      } else if (token == "aten::sub") {
+        output_list.emplace_back(x.sub_(h, const_val));
```

Comment:
As discussed offline, update the docs above to describe what this backend actually implements, which I think it's just support for Modules that implement a sum or subtraction of 2 inputs (i.e. in1 + in2 or in1 - in2)

Reply:
Thanks! Comment added and submitting this PR soon, after resolving CI.